### ISSUE TITLE
In Http implementation add more context to error messages like TimeoutException

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#681]
The messages should contain 
 * the address/port of the peer host (maybe also local address/port)
 * the uri of the request that was being processed when the error occurred

**Comment by ktoso** [comment]
Yep, thanks for opening this ticket. I'll likely pick this up very soon.
